### PR TITLE
[WIP] Upgrade the baseline of lic2021 dialogue competition.

### PR DESCRIPTION
**Pull request by xiemoyuan**
Upgrade the baseline of lic2021 dialogue competition using Paddle2.0

### CODE REVIEW

File: PaddleNLP/examples/dialogue/lic2021/args.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import argparse
+
+
+def parse_args():
```
Comment (ZeyuChen) [inline PaddleNLP/examples/dialogue/lic2021/args.py]: 后续统一下标准，parse_args这里
一个flag一行代码，使用yapf:disable 和yapf:enable 隔开，避免语法检查。减少下代码行数

File: PaddleNLP/examples/dialogue/lic2021/finetune.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+def main(args):
+    paddle.set_device('gpu' if args.n_gpus else 'cpu')
+    world_size = dist.get_world_size()
+    rank = dist.get_rank()
+    if world_size > 1:
+        dist.init_parallel_env()
+
+    vocab = Vocabulary(args.vocab_file)
+
+    model = BaselineModel(
```
Comment (ZeyuChen) [inline PaddleNLP/examples/dialogue/lic2021/finetune.py]: 感觉BaselineModel不是个好名称，是否有一个基础模型名称，如XXTransformer？

File: PaddleNLP/examples/dialogue/lic2021/data.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+from collections import defaultdict
+import paddle.distributed as dist
+from paddle.io import IterableDataset
+
+
+@contextmanager
+def open_file(filename):
+    """Open file."""
+    if filename.endswith(".gz"):
+        fp = gzip.open(filename, "rt")
```
Comment (sserdoubleh) [inline PaddleNLP/examples/dialogue/lic2021/data.py]: add `import gzip`

File: PaddleNLP/examples/dialogue/lic2021/data.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        self.bos_id = vocab.bos_id
+        self.global_rng = np.random.RandomState(seed)
+
+        assert len(self.file_list) > 0, 'There is no files in %s.' % filepattern
+
+    def load_file(self, file_path):
+        with open_file(file_path) as fin:
+            for i, line in enumerate(fin):
+                cols = convert_to_unicode(line).strip().split(";")
+                cols = list(map(lambda x: list(map(int, x.split(" "))), cols))
```
Comment (sserdoubleh) [inline PaddleNLP/examples/dialogue/lic2021/data.py]: 只支持序列化数据吗？

File: PaddleNLP/examples/dialogue/lic2021/finetune.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    valid_dataloader = DataLoader(
+        valid_dataset, return_list=True, batch_size=None)
+
+    lr_scheduler = NoamDecay(1 / (args.warmup_steps * (args.lr**2)),
+                             args.warmup_steps)
+    optimizer = AdamW(
+        learning_rate=lr_scheduler,
+        parameters=model.parameters(),
+        weight_decay=args.weight_decay,
+        grad_clip=nn.ClipGradByGlobalNorm(args.max_grad_norm))
```
Comment (sserdoubleh) [inline PaddleNLP/examples/dialogue/lic2021/finetune.py]: weight decay exclude LN parameters

File: PaddleNLP/examples/dialogue/lic2021/data.py (hunk 3 above)
Comment (xiemoyuan) [reply PaddleNLP/examples/dialogue/lic2021/data.py]: Done. Thanks.

File: PaddleNLP/examples/dialogue/lic2021/data.py (hunk 4 above)
Comment (xiemoyuan) [reply PaddleNLP/examples/dialogue/lic2021/data.py]: 数据读取这边的要求是怎样的呢？给到我这边的数据集是序列化后的，是先运行一个数据转换脚本转成序列化的数据吗？后续需要直接用原始数据作为输入吗？

File: PaddleNLP/examples/dialogue/lic2021/finetune.py (hunk 5 above)
Comment (xiemoyuan) [reply PaddleNLP/examples/dialogue/lic2021/finetune.py]: Done. Thanks.

**Review by ZeyuChen — CHANGES_REQUESTED** [review]
精简下args参数这地方，简化下代码，目前这里他Icahn过了

File: PaddleNLP/examples/dialogue/lic2021/finetune.py (hunk 2 above)
Comment (xiemoyuan) [reply PaddleNLP/examples/dialogue/lic2021/finetune.py]: 原来的名字叫UnifiedTransformer，那我改成这个

File: PaddleNLP/examples/dialogue/lic2021/args.py (hunk 1 above)
Comment (xiemoyuan) [reply PaddleNLP/examples/dialogue/lic2021/args.py]: Done. Thanks.
